lib/adal.js

```diff
@@ -1362,7 +1362,9 @@ var AuthenticationContext = (function () {
 
             if (window.parent === window && !isPopup) {
                 if (self.config.navigateToLoginRequestUrl) {
-                    window.location.href = self._getItem(self.CONSTANTS.STORAGE.LOGIN_REQUEST);
+                    var navigateUrl = self._getItem(self.CONSTANTS.STORAGE.LOGIN_REQUEST);
+                    window.setTimeout(function(){window.location.href = navigateUrl;}, 0);
```

Comment:
I'm not sure I understand why this approach was choosen. Is intent that this will delay the redirecting to `CONSTANTS.STORAGE.LOGIN_REQUEST`? With the underlying issue being that the redirect is happening too early, thus causing the app to send the user back to `login.microsoftonline.com`?

Reply:
This will clear any request pending for login.microsoftonline.com and user will be redirect correctly to CONSTANTS.STORAGE.LOGIN_REQUEST

Reply:
`self.CONSTANTS.STORAGE.LOGIN_REQUEST` should never be set to a `login.microsoftonline.com` url. This url is always set to `window.location.href` when the user is on your application (e.g. `teams.microsoft.com`), right before the redirect to `login.microsoftonline.com`:

- login: https://github.com/AzureAD/azure-activedirectory-library-for-js/blob/3c9ba88a63d07ea081364141946216f7103261f4/lib/adal.js#L225
- acquireToken: https://github.com/AzureAD/azure-activedirectory-library-for-js/blob/3c9ba88a63d07ea081364141946216f7103261f4/lib/adal.js#L791

Reply:
Agree. but somewhere redirect to login.microsoftonline.com is pending which redirect browser to login.microsoftonline.com instead of CONSTANTS.STORAGE.LOGIN_REQUEST.

This approach which make sure to clear any request pending for login.microsoftonline.com and user will be redirect correctly to CONSTANTS.STORAGE.LOGIN_REQUEST